api/file.go

```diff
@@ -612,16 +614,42 @@ func migrateFilenamesToFileInfos(post *model.Post) []*model.FileInfo {
 				continue
 			}
 
-			if result := <-Srv.Store.FileInfo().Save(info); result.Err != nil {
-				l4g.Error(utils.T("api.file.migrate_filenames_to_file_infos.save_file_info.app_error"), post.Id, info.Id, filename, result.Err)
-				continue
-			}
-
-			fileIds = append(fileIds, info.Id)
 			infos = append(infos, info)
 		}
 	}
 
+	// Lock to prevent only one migration thread from trying to update the post at once, preventing duplicate FileInfos from being created
+	fileMigrationLock.Lock()
+	defer fileMigrationLock.Unlock()
```

Comment:
have you tried this with a bunch of load on the server? this actually makes me nervous